File: solutions/_946_validate_stack_sequences/solution.java

package _946_validate_stack_sequences;

import java.util.*;

class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        var stack = new Stack<Integer>();
        int i = 0;
        for (int val : pushed) {
            stack.push(val);
            while (stack.size() > 0 && stack.peek() == popped[i]) {
                stack.pop();
                i++;
            }
        }
        return stack.size() == 0;
    }
}

/*
 * https://leetcode.com/problems/validate-stack-sequences/
 */

File: solutions/_946_validate_stack_sequences/solution.js
/**
 * @param {number[]} pushed
 * @param {number[]} popped
 * @return {boolean}
 */
var validateStackSequences = function (pushed, popped) {
  const stack = [];
  let i = 0;

  for (const val of pushed) {
    stack.push(val);
    while (stack.length && stack[stack.length - 1] === popped[i]) {
      stack.pop();
      i++;
    }
  }

  return stack.length === 0;
};

module.exports = { validateStackSequences };

/*
https://leetcode.com/problems/validate-stack-sequences/
*/
